src/Serializer/AbstractItemNormalizer.php

```diff
@@ -147,14 +148,38 @@ protected function extractAttributes($object, $format = null, array $context = [
      */
     protected function getAllowedAttributes($classOrObject, array $context, $attributesAsString = false)
     {
+        $allowExtraAttributes = $context[self::ALLOW_EXTRA_ATTRIBUTES] ?? $this->defaultContext[self::ALLOW_EXTRA_ATTRIBUTES];
+        if (!$this->classMetadataFactory) {
+            if (!$allowExtraAttributes) {
+                throw new LogicException(sprintf('A class metadata factory must be provided in the constructor when setting "%s" to false.', self::ALLOW_EXTRA_ATTRIBUTES));
+            }
+
+            return false;
+        }
+
         $options = $this->getFactoryOptions($context);
         $propertyNames = $this->propertyNameCollectionFactory->create($context['resource_class'], $options);
 
         $allowedAttributes = [];
+        $groups = $context[self::GROUPS] ?? false;
+        $attributesInGroup = [];
+
+        if (false !== $groups) {
+            $classMetaData = $this->classMetadataFactory->getMetadataFor($classOrObject);
+            foreach ($classMetaData->getAttributesMetadata() as $attributeMetadata) {
+                $name = $attributeMetadata->getName();
+
+                if (array_intersect($attributeMetadata->getGroups(), $groups)) {
+                    $attributesInGroup[] = $name;
+                }
+            }
+        }
```

Comment:
looks legit, not sure about having a `$groups` variable that can be either a boolean either an array...

Reply:
I took the idea for `$groups` being false from [Symfonys AbstractNormalizer](https://github.com/symfony/serializer/blob/master/Normalizer/AbstractNormalizer.php#L281), but I will change this, if this kind of style is not wished in the api-platform. @soyuka